### ISSUE TITLE
Week methods only work with local translator, not default locale

### ISSUE THREAD

**Issue by hackel** [briannesbitt/Carbon#1967]
Hello,

I encountered an issue with the following code:
```php
echo Carbon::getLocale(); // en
echo Carbon::now()->locale(); // en
echo Carbon::now()->startOfWeek()->dayName; // Monday
Carbon::setLocale('en_US');
echo Carbon::now()->startOfWeek()->dayName; // Monday

echo Carbon::now()->locale('en')->startOfWeek()->dayName; // Sunday
```
Trying to figure out all of the different (apparently competing?) localization settings makes my head spin.  The issue seems to be that "firstWeekDay" will only ever be determined by a local translator, not the default translator:

Carbon\Traits\Localization\Date:928
```php
case $name === 'firstWeekDay':
    return $this->localTranslator ? ($this->getTranslationMessage('first_day_of_week') ?? 0) : static::getWeekStartsAt();
```

I'm not sure what the point of `static::$weekStartsAt` is...backward compatibility?  It conflicts with what the default translator is supposed to return.

As far as I can tell, the only way to solve this is to use the deprecated methods `setWeekStartsAt()` AND `setWeekEndsAt()` (which for some bizarre reason is not calculated automatically), or explicitly include the `locale()` call every single time I use these methods (even though it already is the default).  Am I missing something obvious here?

I would propose removing the default values for `static::$weekStartsAt/EndsAt`, and changing the getters to only use them if they are explicitly set:
```php
case $name === 'firstWeekDay':
    return static::getWeekStartsAt() ?? $this->getTranslationMessage('first_day_of_week', null, 0);
```

This is a simple change and I would be happy to submit a PR if acceptable, but it would technically be breaking, even though it's a bugfix in my opinion.

Carbon version: 2.28.0

PHP version: 7.4.0

I expected to get: (with default 'en' or 'en_US' locale)

```
echo Carbon::now()->startOfWeek()->dayName; // Sunday
```

But I actually get:

```
echo Carbon::now()->startOfWeek()->dayName; // Monday
```

Thanks!

**Comment by kylekatarnls** [comment]
Hello,

This is a legacy. `startOfWeek` exists for a long time before `first_day_of_week` was supported in different locales.

At this time Monday as the standard start of week was the default value for `startOfWeek` and when the I18N was added, default locale `en` has been chosen as `diffForHumans` returned English text.

So changing this behavior would not be a bug fix but really a breaking changes.

Many users user `startOfWeek` without using actually using I18N (so they are implicitly on the default locale `en`), and Monday is the start of week for the very majority of people in the world.

This change could break apps that legitimately rely on this. An other solution must be found or it should wait a version 3.

**Comment by kylekatarnls** [comment]
In Carbon 3, the default locale will be [en_001](http://www.localeplanet.com/icu/en-001/index.html) “English (World)”.

And default start of week and every calendar/format features will be aligned on this locale as a standard. Then our "en" locale will be identical to the "en_001" and so we will keep "en" as default locale (as in Laravel config) but it will now mean "en_001" and no longer "en_US" as in Carbon 2, and so default start of week will match the default locale. There is basically no wording difference between en_001 and en_US, so it should be a smooth change.

⚠️ The date formats still have to be checked. Some sources gives dd/MM, some other MM/dd so I have to clarify which should be considered as the standard.

**Comment by kylekatarnls** [comment]
While working on this for 3.0 I actually realized I could use an opt-in mode to enable the behavior of 3.0 in the next 2.35 version.

With:
```php
Carbon::setWeekStartsAt('auto');
Carbon::setWeekEndsAt('auto');
```

`startOfWeek()` will rely on the day in the current culture (Sunday as start of en or en_US)

Open PR for this fix: #2090

**Comment by kylekatarnls** [comment]
This can be tested requiring `"nesbot/carbon": "2.x-dev",`
